packages/aws-cdk-lib/aws-ec2/lib/vpc.ts

```diff
@@ -1879,7 +2073,12 @@ export class Subnet extends Resource implements ISubnet {
       cidrBlock: props.cidrBlock,
       availabilityZone: props.availabilityZone,
       mapPublicIpOnLaunch: props.mapPublicIpOnLaunch,
+      ipv6CidrBlock: props.ipv6CidrBlock,
+      assignIpv6AddressOnCreation: props.assignIpv6AddressOnCreation,
```

Comment:
Also set `private_dns_name_options_on_launch`. For IPv6-only subnets, this is automatically enabled, however for dual-stack the default is not, which means they will get an IPv4 internal DNS name but not IPv6. Note this only applies to internal DNS names.

This is not yet fully mapped to CloudFormation, so you just need to pass in an array of any with the known properties.

Set: EnableResourceNameDnsAAAARecord = true, EnableResourceNameDnsARecord = true, HostnameType = 'resource-name'.

This will mean that machines created on the dual stack subnet will have an machine name based on the instance (not the IPv4 address) and be accessible via internal DNS by that name.

By setting for both v4 & v6 that single internal DNS reference (as instance) will work for both IPv4 and IPv6, i.e. a single internal host name.